bezier_library/include/bezier_library/bezier_params_observer.hpp

```diff
@@ -0,0 +1,124 @@
+                     std::numeric_limits<int>::max());
+  }
+
+  virtual ~BezierParamsObserver()
+  {
+  }
+
+  std::string name()
+  {
+    return "BezierParamsObserver";
+  }
+
+  void inline setCurrentGrindingDiskWidth(const double current_grinding_disk_machining_width)
+  {
+    grinding_disk_machining_width_.first = grinding_disk_machining_width_.second;
+    grinding_disk_machining_width_.second = current_grinding_disk_machining_width;
+  }
+
+  void inline setCurrentCoveringPercentage(const unsigned current_covering_percentage)
+  {
+    covering_percentage_.first = covering_percentage_.second;
+    covering_percentage_.second = current_covering_percentage;
+  }
+
+  void inline setCurrentExtricationRadius(const double current_extrication_radius)
+  {
+    extrication_radius_.first = extrication_radius_.second;
+    extrication_radius_.second = current_extrication_radius;
+  }
+
+  void inline setCurrentSlicingDirection(const Eigen::Vector3d current_slicing_orientation)
+  {
+    slicing_orientation_.first = slicing_orientation_.second;
+    slicing_orientation_.second = current_slicing_orientation;
+  }
+
+  void inline setCurrentLeanAngle(const double lean_angle)
+  {
+    lean_angle_.first = lean_angle_.second;
+    lean_angle_.second = lean_angle;
+  }
+
+  void inline setCurrentAxisOfRotation(const int axis_of_rotation)
+  {
+    axis_of_rotation_.first = axis_of_rotation_.second;
+    axis_of_rotation_.second = axis_of_rotation;
+  }
+
+  void setCurrentParams(const double grinding_disk_machining_width, const unsigned covering_percentage,
+                        const double extrication_radius, const Eigen::Vector3d slicing_orientation,
+                        const double lean_angle, const int axis_of_rotation)
+  {
+    setCurrentGrindingDiskWidth(grinding_disk_machining_width);
+    setCurrentCoveringPercentage(covering_percentage);
+    setCurrentExtricationRadius(extrication_radius);
+    setCurrentSlicingDirection(slicing_orientation);
+    setCurrentLeanAngle(lean_angle);
+    setCurrentAxisOfRotation(axis_of_rotation);
+  }
+
+  bool inline mustComputeDilation()
+  {
+    return extrication_radius_.first != extrication_radius_.second;
+  }
+
+  bool inline mustGenerateGrindingTrajectories()
+  {
+    return grinding_disk_machining_width_.first != grinding_disk_machining_width_.second
+        || covering_percentage_.first != covering_percentage_.second
+        || slicing_orientation_.first != slicing_orientation_.second || lean_angle_.first != lean_angle_.second
+        || axis_of_rotation_.first != axis_of_rotation_.second;
+  }
+
+  bool inline mustGenerateExtricationTrajectories()
+  {
+    return mustComputeDilation() || mustGenerateGrindingTrajectories();
+  }
+
+  bool inline mustRecomputeTrajectoryVector()
+  {
+    return mustGenerateExtricationTrajectories() || mustGenerateGrindingTrajectories();
+  }
+
+private:
+  // New Parameters
```

Comment:
// Old / Current parameters (I don't remember the sequence)
Document which one is the old/current